db/db_impl/db_impl_files.cc

```diff
@@ -318,9 +318,12 @@ void DBImpl::PurgeObsoleteFiles(JobContext& state, bool schedule_only) {
   // We may ignore the dbname when generating the file names.
   const char* kDumbDbName = "";
   for (auto& file : state.sst_delete_files) {
-    candidate_files.emplace_back(
-        MakeTableFileName(kDumbDbName, file.metadata->fd.GetNumber()),
-        file.path);
+    std::string table_file_name =
+        MakeTableFileName(kDumbDbName, file.metadata->fd.GetNumber());
```

Comment:
I think a better fix is to avoid this kDumbDbName hack. Instead, creating a new function like MakeTableFileName() which doesn't return the leading "/".